Sources/TuistSupport/Utils/DeveloperEnvironment.swift

```diff
@@ -4,24 +4,31 @@ import TSCBasic
 public protocol DeveloperEnvironmenting {
     /// Returns the derived data directory selected in the environment.
     var derivedDataDirectory: AbsolutePath { get }
+    
+    /// Returns the system's architecture.
+    var architecture: MacArchitecture { get }
 }
 
 public final class DeveloperEnvironment: DeveloperEnvironmenting {
     /// Shared instance to be used publicly.
     /// Since the environment doesn't change during the execution of Tuist, we can cache
     /// state internally to speed up future access to environment attributes.
-    public static let shared: DeveloperEnvironmenting = DeveloperEnvironment()
+    public static var shared: DeveloperEnvironmenting = DeveloperEnvironment()
```

Comment:
Can this be changed to `public internal(set) var ...`? AFAICT, we only need to change this for test purposes (where, by using `@testable` we have internal access) - but it'd be great to ensure that in modules that use this component, you cannot actually change the `shared` instance